sdk/identity/azure-identity/tests/test_shared_cache_credential.py

```diff
@@ -800,6 +800,44 @@ def mock_send(request, **_):
     assert transport.send.called
 
 
+def test_client_capabilities():
+    """the credential should configure MSAL for capability CP1 (ability to handle claims challenges)"""
+
+    record = AuthenticationRecord("tenant-id", "client_id", "authority", "home_account_id", "username")
+    transport = Mock(send=Mock(side_effect=Exception("this test mocks MSAL, so no request should be sent")))
+    credential = SharedTokenCacheCredential(transport=transport, authentication_record=record, _cache=TokenCache())
+
+    with patch(SharedTokenCacheCredential.__module__ + ".PublicClientApplication") as PublicClientApplication:
+        credential._initialize()
+
+    assert PublicClientApplication.call_count == 1
+    _, kwargs = PublicClientApplication.call_args
+    assert kwargs["client_capabilities"] == ["CP1"]
+
+
+def test_claims_challenge():
+    """get_token should pass any claims challenge to MSAL token acquisition APIs"""
+
+    expected_claims = '{"access_token": {"essential": "true"}'
+
+    record = AuthenticationRecord("tenant-id", "client_id", "authority", "home_account_id", "username")
+
+    msal_app = Mock()
+    msal_app.get_accounts.return_value = [{"home_account_id": record.home_account_id}]
+    msal_app.acquire_token_silent_with_error.return_value = dict(
+        build_aad_response(access_token="**", id_token=build_id_token())
```

Comment:
Does the `SharedTokenCacheCredential` not have an `id_token_claims` portion of the return value (like [this one](https://github.com/Azure/azure-sdk-for-python/pull/16323/files?file-filters%5B%5D=.py#diff-310b4541094eb834960847f01d92d60f5f9123fb62d181388e74d54ae6493d46R170)) because it's the only credential of these that doesn't subclass `InteractiveCredential`?

Reply:
Yes. That's relevant because MSAL token acquisition methods return a dict containing both the raw ID token and its claims. `InteractiveCredential` [uses these claims to build an `AuthenticationRecord`](https://github.com/Azure/azure-sdk-for-python/blob/79a64bb9d40610d1696819a2595a49d18d7a7ad6/sdk/identity/azure-identity/azure/identity/_internal/interactive.py#L51). `SharedTokenCacheCredential` does not, because it already has that information.